python/ray/autoscaler/_private/resource_demand_scheduler.py

```diff
@@ -161,7 +176,8 @@ def get_nodes_to_launch(
         if self.is_legacy_yaml():
             # When using legacy yaml files we need to infer the head & worker
             # node resources from the static node resources from LoadMetrics.
-            self._infer_legacy_node_resources_if_needed(max_resources_by_ip)
+            self._infer_legacy_node_resources_if_needed(
```

Comment:
Probably put of scope of this PR, but I think it would be cleaner if we handled legacy stuff on yaml load (convert everything to non-legacy form there and there).

Reply:
I think this part has to be done at runtime. The better thing to do at this point (in another PR) may be to delete all the legacy logic -- it's been deprecated for long enough... though obviously the correct thing to do is to first issue warning s that things will cease to work in Ray 1.x.0.